src/wallet/wallet.cpp

```diff
@@ -1155,6 +1155,31 @@ void CWallet::BlockConnected(const std::shared_ptr<const CBlock>& pblock, const
     for (size_t i = 0; i < pblock->vtx.size(); i++) {
         SyncTransaction(pblock->vtx[i], pindex, i);
     }
+
+    // The GUI expects a NotifyTransactionChanged when a coinbase tx
+    // which is in our wallet moves from in-the-best-block to
+    // 2-confirmations. We do that here.
```

Comment:
Updated the comment, the GUI simply doesnt display such transactions until they are buried.